server/src/main/kotlin/delta/codecharacter/server/daily_challenge/DailyChallengeService.kt

```diff
@@ -23,34 +31,45 @@ class DailyChallengeService(
         return period.toDays().toInt()
     }
 
-    fun getDailyChallengeByDate(): DailyChallengeGetRequestDto {
-
+    fun getDailyChallengeByDate(): DailyChallengeEntity {
         val dc =
             dailyChallengeRepository.findByDay(findNumberOfDays()).orElseThrow {
                 throw CustomException(org.springframework.http.HttpStatus.BAD_REQUEST, "Invalid Request")
             }
+        return dc
+    }
+
+    fun getDailyChallengeByDateForUser(userId: UUID): DailyChallengeGetRequestDto {
+        val user = publicUserService.getPublicUser(userId)
+        val dc = getDailyChallengeByDate()
         return DailyChallengeGetRequestDto(
             challName = dc.challName,
             chall = dc.chall,
             challType = dc.challType,
             description = dc.description,
-            completionStatus = false
+            completionStatus = user.dailyChallengeHistory.containsKey(dc.id)
         )
     }
 
-    fun completeDailyChallenge(gameEntity: GameEntity) {
-        val (_, coinsUsed, destruction, status, matchId) = gameEntity
-    /*
-     * if destruction >= 60 --> completed --> usr can not play
-     * else not-completed user can play
-     *
-     * score = base-score + (value depending on coinsUsed and destruction %)
-     * base-score will get reduced as player starts solving challenge
-     * base-score = base-score - rv (rv-> reducing value) (like CTF style)
-     *
-     * Corresponding LeaderBoard updates
-     *
-     * Scheduling for isDailyChallengeComplete -> false by 24hrs
-     */
+    fun completeDailyChallenge(gameEntity: GameEntity, userId: UUID): DailyChallengeMatchVerdictEnum {
+        val (_, coinsUsed, destruction, _, _) = gameEntity
+        if (gameEntity.status == GameStatusEnum.EXECUTE_ERROR)
+            return DailyChallengeMatchVerdictEnum.FAILURE
+        val dc = dailyChallengeRepository.findByDay(findNumberOfDays()).get()
```

Comment:
Instead you can make use of the `getDailyChallengeByDate()` function already declared in `DailyChallengeService.kt` .